File: examples/src/Slides/Layout/SetSizeAndType.cs
﻿using Aspose.slides.Export

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
when the project is build. Please check https:#docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http:#www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:#www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.slides.Examples.CSharp.slides.Layout
{
    class SetSizeAndType
    {
        public static void Run()
        {
            #ExStart:SetSizeAndType
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Slides_Presentations_Layout()

            # Instantiate a Presentation object that represents a presentation file 
            Presentation presentation = new Presentation(dataDir + "AccessSlides.pptx")
            Presentation auxPresentation = new Presentation()

            slide = presentation.slides[0]

            # Set the slide size of generated presentations to that of source
            auxPresentation.SlideSize.SetSize(presentation.SlideSize.type,SlideSizeScaleType.EnsureFit)

            auxPresentation.slides.insert_clone(0, slide)
            auxPresentation.slides.remove_at(0)
            # Save Presentation to disk
            auxPresentation.save(dataDir + "Set_Size&Type_out.pptx", slides.export.SaveFormat.PPTX)
            #ExEnd:SetSizeAndType
        }
    }
}

File: examples/src/Slides/layout/set_size_and_type.py
import aspose.slides as slides


def set_size_and_type(global_opts):
    # Instantiate a Presentation object that represents a presentation file
    with slides.Presentation(global_opts.data_dir + "welcome-to-powerpoint.pptx") as presentation:
        with slides.Presentation() as aux_presentation:
            slide = presentation.slides[0]

            # Set the slide size of generated presentations to that of source
            aux_presentation.slide_size.set_size(presentation.slide_size.type, slides.SlideSizeScaleType.ENSURE_FIT)

            aux_presentation.slides.insert_clone(0, slide)
            aux_presentation.slides.remove_at(0)
            # Save Presentation to disk
            aux_presentation.save(global_opts.out_dir + "layout_slide_size_out.pptx", slides.export.SaveFormat.PPTX)
